app/src/main/java/com/example/hackernews/view/activities/MainActivity.kt

```diff
@@ -54,14 +59,33 @@ class MainActivity :
     }
 
     private fun bindViewModel() {
-        viewModel.allNews.observe(this, { news ->
+        viewModel.news.observe(this, { news ->
```

Comment:
Use this to send all kinds of news to the view. So for example in ViewModel when you fetch latest news `news.value = latestNews` or when you get savedNews `news.value = savedNews` etc. that way you won't have issue of mixing different types of news.